client/lib/application/src/application-lifecycle/ApplicationLifeCycleComponent.jsx

```diff
@@ -0,0 +1,140 @@
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const versions = nextProps.aliceStore.instanceCountData.map( e => {return {id: e.version_id}});
+        this.setState({
+            versions,
+            selectedVersions: versions
+        });
+    }
+
+    // handler functions
+
+    handleStartDatePicked(date) {
+        this.handleDateChanged(moment(date).startOf('day').toDate(), this.state.endDate);
+    }
+
+    handleEndDatePicked(date) {
+        this.handleDateChanged(this.state.startDate, moment(date).endOf('day').toDate());
+    }
+
+    handleDateChanged(startDate, endDate) {
+        this.setState({
+            startDate,
+            endDate,
+            brushExtentStartDate: startDate,
+            brushExtentEndDate: endDate
+        });
+        this.props.aliceActions.fetchInstanceCount(this.props.params.applicationId, startDate, endDate);
+    }
+
+    handleBrushChanged([brushExtentStartDate, brushExtentEndDate]) {
+        this.setState({
+            brushExtentStartDate,
+            brushExtentEndDate
+        });
+    }
+
+    handleRemoveVersion(versionId) {
+        const { selectedVersions } = this.state;
+        this.handleVersionsSelect(selectedVersions.filter(v => v.id != versionId));
+    }
+
+    handleVersionsSelect(param) {
+        this.setState({selectedVersions: param});
+    }
+
+    handleVersionReset() {
+        this.setState({selectedVersions: this.state.versions});
+    }
+
+    // render function
+
+    render() {
+        return (
+            <ApplicationLifeCycle
+                applicationId        = {this.props.params.applicationId}
+                versions             = {this.state.versions}
+                selectedVersions     = {this.state.selectedVersions}
+                onVersionsSelect     = {this.handleVersionsSelect}
+                onVersionReset       = {this.handleVersionReset}
+                onStartDatePicked    = {this.handleStartDatePicked}
+                onEndDatePicked      = {this.handleEndDatePicked}
+                onBrushChanged       = {this.handleBrushChanged}
+                onRemoveVersion      = {this.handleRemoveVersion}
+                startDate            = {this.state.startDate}
+                endDate              = {this.state.endDate}
+                brushExtentStartDate = {this.state.brushExtentStartDate}
+                brushExtentEndDate   = {this.state.brushExtentEndDate}
+                {...this.props} />
+        );
+    }
+}
+
+ApplicationLifecycleHandler.displayName = 'ApplicationLifecycleHandler';
+
+ApplicationLifecycleHandler.propTypes = {
+    aliceActions: React.PropTypes.shape({
+        fetchInstanceCount: React.PropTypes.func
+    }).isRequired,
+    params: React.PropTypes.shape({
+        applicationId: React.PropTypes.string
+    }).isRequired
+};
+
+function mapStateToProps(state) {
+    return {
+        aliceStore: state.alice,
+        kioStore: bindGettersToState(state.kio, KioGetter)
```

Comment:
I'd like to not have this pattern in new code, as it was just introduced for quicker refactoring when switching from Flummox to Redux. Please pass relevant data directly as props. (Same goes for actions, no need for `bindActionCreators` — pass the actions as props.)